akka-http-core/src/main/scala/akka/http/impl/engine/http2/ProtocolSwitch.scala

```diff
@@ -4,28 +4,29 @@
 
 package akka.http.impl.engine.http2
 
-import akka.NotUsed
 import akka.annotation.InternalApi
-import akka.http.impl.engine.server.HttpAttributes
+import akka.http.impl.engine.http2.Http2.HttpImplementation
+import akka.http.impl.engine.server.{ HttpAttributes, ServerTerminator }
 import akka.stream.ActorAttributes.Dispatcher
 import akka.stream.TLSProtocol.{ SessionBytes, SessionTruncated, SslTlsInbound, SslTlsOutbound }
 import akka.stream._
 import akka.stream.scaladsl.Flow
-import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
+import akka.stream.stage.{ GraphStageLogic, GraphStageWithMaterializedValue, InHandler, OutHandler }
 
 import javax.net.ssl.SSLException
+import scala.concurrent.{ Future, Promise }
 
 /** INTERNAL API */
 @InternalApi
 private[http] object ProtocolSwitch {
-  type HttpServerFlow = Flow[SslTlsInbound, SslTlsOutbound, NotUsed]
+  type HttpServerFlow = Flow[SslTlsInbound, SslTlsOutbound, Future[ServerTerminator]]
```

Comment:
(since it's now only used in only one place anymore, perhaps we should remove this `type` and just type it out below)